lib/bedrock.js

```diff
@@ -368,6 +368,37 @@ function _setupUncaughtExceptionHandler(logger, logPrefix) {
   }
 }
 
+function _setupSignalHandler(logger, logPrefix) {
+  const SIGNALS = {
+    /*
+      The SIGHUP (“hang-up”) signal is used to report that the
+      user’s terminal is disconnected, perhaps because a network
+      or telephone connection was broken.
+    */
+    SIGHUP: 1,
+    /*
+      The SIGINT (“program interrupt”) signal is sent
+      when the user types the INTR character (normally C-c).
+    */
+    SIGINT: 2,
+    /*
+      The SIGTERM signal is a generic signal used to cause
+      program termination. Unlike SIGKILL, this signal can
+      be blocked, handled, and ignored. It is the normal way
+      to politely ask a program to terminate.
+    */
+    SIGTERM: 15
+  };
+
+  Object.keys(SIGNALS).forEach(signal => {
+    process.on(signal, async () => {
+      logger.critical(`${logPrefix} received signal [${signal}].`);
+      await events.emit('bedrock.exit');
+      process.exit();
```

Comment:
I didn't look into this, but does the logging system need some sort of final flush before `process.exit()` is called?

Reply:
https://github.com/winstonjs/winston#awaiting-logs-to-be-written-in-winston

https://nodejs.org/docs/latest-v14.x/api/stream.html#stream_event_finish

I guess the suggestion is  that in the bedrock exit handler.  We would register an even listener for `logger.on('finish')`, and then log "something" from within the exit handler itself followed by a `logger.end()`, and then when we see the `finish` event fire after that, we know the stream is done.

Reply:
Added issue #80 to handle this later.